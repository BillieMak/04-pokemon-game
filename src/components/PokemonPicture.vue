<template>
  <h3>Pokemon Picture</h3>
  <div class="pokemon-container">
    <img class="hidden-pokemon" :src="imgSrc" alt="Pokemon">
    <img v-if="showPokemon" class="fade-in" :src="imgSrc" alt="Pokemon">
  </div>
</template>

<script>
export default {
  name: 'PokemonPicture',
  props: {
    pokemonId: {
      type: Number,
      required: true
    },
    showPokemon: {
      type: Boolean,
      required: true,
      default: false
    }
  },
  computed: {
    imgSrc() {
      return `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${this.pokemonId}.svg`
    }
  }
}
</script>

<style scoped>
.pokemon-container {
  height: 200px;
  animation: float 3s ease-in-out infinite;
}

img {
  height: 200px;
  position: absolute;
  width: 100vw;
  margin-top: 30px;
  margin-left: -50vw;
  user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-drag: none;
  -webkit-user-select: none;


}

.hidden-pokemon {
  filter: brightness(0);
}

@keyframes float{
  0% {
    transform: translateY(0px);
  }
  50% {
    transform: translateY(-40px);
  }
  100% {
    transform: translateY(0px);
  }
}

@media (min-width: 769px) and (max-width: 1024px) {
  .image-container {
    width: 50%;
  }
}

@media (min-width: 1025px) {
  .image-container {
    width: 30%;
  }
}
</style>