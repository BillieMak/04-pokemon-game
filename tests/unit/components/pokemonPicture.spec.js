import { shallowMount } from "@vue/test-utils";
import PokemonPicture from "@/components/PokemonPicture";

describe("Pruebas en Pokemon Picture", () => {
  test("Debe hacer match con el snapshot", () => {
    const wrapper = shallowMount(PokemonPicture, {
      props: {
        pokemonId: 1,
        showPokemon: false,
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("Debe mostrar la imagen oculta y el pokemon 100", () => {
    const wrapper = shallowMount(PokemonPicture, {
      props: {
        pokemonId: 100,
        showPokemon: false,
      },
    });
    const imgs = wrapper.findAll("img");
    expect(imgs).toHaveLength(1);

    const [img1] = imgs;

    expect(img1.classes("hidden-pokemon")).toBeTruthy();
  });

  test("Debe mostrar el pokemon si showPokemon:true", () => {
    const wrapper = shallowMount(PokemonPicture, {
      props: {
        pokemonId: 100,
        showPokemon: true,
      },
    });
    const imgs = wrapper.findAll("img");
    expect(imgs).toHaveLength(2);

    const [img1, img2] = imgs;
    expect(img1.classes("hidden-pokemon")).toBeTruthy();
    expect(img2.classes("show-pokemon")).toBeFalsy();
    expect(img2.classes("fade-in")).toBeTruthy();
    expect(img2.attributes("src")).toBe("https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg");
  });
});
