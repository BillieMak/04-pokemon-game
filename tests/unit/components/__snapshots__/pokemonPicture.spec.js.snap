// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en Pokemon Picture Debe hacer match con el snapshot 1`] = `
<h3>Pokemon Picture</h3>
<div class="pokemon-container"><img class="hidden-pokemon" src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg" alt="Pokemon">
  <!--v-if-->
</div>
`;
